templates/shop/order/thank_you/content/buttons/change_payment_method.html.twig

```diff
@@ -0,0 +1,7 @@
+{% import '@SyliusShop/shared/buttons.html.twig' as buttons %}
+{% set order = hookable_metadata.context.order %}
+{% set lastPayment = order.payments.last() %}
+
+{% if lastPayment.state != constant('Sylius\\Component\\Payment\\Model\\PaymentInterface::STATE_COMPLETED') %}
```

Comment:
```suggestion
{% set lastPayment = order.getLastPayment(constant('Sylius\\Component\\Payment\\Model\\PaymentInterface::STATE_COMPLETED')) %}

{% if lastPayment is not null %}
```

Reply:
This is not the same logic. In the first case, it was checking whether the last payment has a status other then `completed`, whereas in your case, it checks all payments.